src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+import { Constants, DistributedSettingUpdateType } from "./constants";
+import { TelemetryClient } from "./telemetryClient";
+import { Utility } from "./utility";
+import iothub = require("azure-iothub");
+import uuid = require("uuid");
+import { DeviceItem } from "./Model/DeviceItem";
+import { SamplingModeItem } from "./Model/SamplingModeItem";
+import { DistributedTracingLabelNode } from "./Nodes/DistributedTracingLabelNode";
+import { DistributedTracingSettingNode } from "./Nodes/DistributedTracingSettingNode";
+
+export class DistributedTracingManager extends BaseExplorer {
+    constructor(outputChannel: vscode.OutputChannel) {
+        super(outputChannel);
+    }
+
+    public async updateDistributedTracingSetting(node, updateType?: DistributedSettingUpdateType) {
+        let iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);
+        if (!iotHubConnectionString) {
+            return;
+        }
+
+        if (!updateType) {
+            updateType = DistributedSettingUpdateType.All;
+        }
+
+        let deviceIds: string[] = [];
+        if (!node) {
+            const deviceIdList = await Utility.getNoneEdgeDeviceIdList(iotHubConnectionString);
+            const deviceItemList = deviceIdList.map((deviceId) => new DeviceItem(deviceId, null, null, null, null));
+            let selectedDevices: DeviceItem[] = await vscode.window.showQuickPick(
+                deviceItemList,
+                { placeHolder: "Select device...", ignoreFocusOut: true, canPickMany: true },
+            );
+
+            if (selectedDevices !== undefined && selectedDevices.length > 0) {
+                deviceIds = selectedDevices.map((deviceItem) => deviceItem.deviceId);
+            }
+        } else {
+            deviceIds = [node.deviceNode.deviceId];
+        }
+
+        if (deviceIds.length === 0) {
+            return;
+        }
+
+        this._outputChannel.show();
+        await this.updateDistributedTracingSettingForDevices(deviceIds, iotHubConnectionString, updateType);
+
+        if (node instanceof DistributedTracingLabelNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node);
+        } else if (node instanceof DistributedTracingSettingNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node.parent);
+        } else {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh");
+        }
+    }
+
+    public async updateDistributedTracingSettingForDevices(deviceIds: string[], iotHubConnectionString: string, updateType: DistributedSettingUpdateType) {
+        let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+        let mode: boolean = undefined;
+        let samplingRate: number = undefined;
+        let twin;
+
+        if (deviceIds.length === 1) {
+            twin = await Utility.getTwin(registry, deviceIds[0]);
```

Comment:
If network is not good, this may take some time. Better to have some UI hint

Reply:
Use window.withProgress to hint user of the progress.